repo2docker/app.py

```diff
@@ -393,20 +405,25 @@ def initialize(self, argv=None):
         if args.appendix:
             self.appendix = args.appendix
 
+        self.repo = args.repo
+        self.ref = args.ref
+        # if the source exists locally we don't want to delete it at the end
         if os.path.exists(args.repo):
-            # Let's treat this as a local directory we are building
-            self.repo_type = 'local'
-            self.repo = args.repo
-            self.ref = None
             self.cleanup_checkout = False
-            if args.editable:
-                self.volumes[os.path.abspath(args.repo)] = '.'
         else:
-            self.repo_type = 'remote'
-            self.repo = args.repo
-            self.ref = args.ref
             self.cleanup_checkout = args.clean
 
+        # user wants to mount a local directory into the container for
+        # editing
+        if args.editable:
+            if os.path.isdir(args.repo):
```

Comment:
checking explicitly for it to be a directory not just a path because we might have content providers that can provide content from a local `something.zip` file, which we couldn't mount in editable mode

Reply:
Your comment seems worthy of a brief comment on the `isdir` check